mender-auth/cli/actions.cpp

```diff
@@ -45,21 +45,12 @@ shared_ptr<MenderKeyStore> KeystoreFromConfig(
 	string pem_file;
 	string ssl_engine;
 
-	// TODO: review and simplify logic as part of MEN-6668. See discussion at:
-	// https://github.com/mendersoftware/mender/pull/1378#discussion_r1317185066
-	if (config.https_client.key != "") {
-		pem_file = config.https_client.key;
-		ssl_engine = config.https_client.ssl_engine;
-		static_key = cli::StaticKey::Yes;
-	}
 	if (config.security.auth_private_key != "") {
 		pem_file = config.security.auth_private_key;
 		ssl_engine = config.security.ssl_engine;
 		static_key = cli::StaticKey::Yes;
-	}
-	if (config.https_client.key == "" && config.security.auth_private_key == "") {
+	} else {
 		pem_file = config.paths.GetKeyFile();
-		ssl_engine = config.https_client.ssl_engine;
 		static_key = cli::StaticKey::No;
 	}
```

Comment:
I think it is not right to associate `StaticKey::No` to everything but `config.security.auth_private_key != ""`. In other words, this sets no static keys always for the https configuration, potentially working incorrectly with hardware crypto for example (it will work fine as-is, because we don't have code to attempt regenerating the https key anyway, but still the design is wrong I think).

Sorry for opening now this melon, but I think this should be a filed exposed in the configuration. It is just too much of a guess to figure this out for the user.

Reply:
By the way, thank you for addressing this! :muscle:

Reply:
Discussed f2f, keeping it as it is.